crates/common/log_manager/src/log_utils.rs

```diff
@@ -153,6 +160,30 @@ pub fn filter_logs_path_on_metadata(
     }
 }
 
+fn temp_filename(logfile_path: &Path) -> Result<PathBuf, LogRetrievalError> {
+    let file_name = logfile_path
+        .file_name()
+        .ok_or_else(|| FileError::InvalidFileName {
+            path: logfile_path.to_path_buf(),
+            source: anyhow!("Does not name a valid file"),
```

Comment:
I see that this forced you to use `anyhow` as a dependency. Using `anyhow` in the `source` field of `FileError::InvalidFileName` was a mistake. As a short term solution, I'd suggest that you use `PathsError` here as well. Feel free to add one `InvalidFileName` variant as well.